shared/reducers/fs.tsx

```diff
@@ -148,6 +146,22 @@ export default Container.makeReducer<FsGen.Actions, Types.State>(initialState, {
         )
       draftState.pathItems.set(path, newPathItem)
     })
+
+    // Remove Rename edits that are for path items that don't exist anymore in
+    // case when/if a new item is added later the edit causes confusion.
+    const newEntries = [...draftState.edits.entries()].filter(([_, edit]) => {
+      if (edit.type !== Types.EditType.Rename) {
+        return true
+      }
+      const parent = Constants.getPathItem(draftState.pathItems, edit.parentPath)
+      if (parent.type === Types.PathType.Folder && parent.children.has(edit.name)) {
+        return true
+      }
+      return false
+    })
+    if (newEntries.length !== draftState.edits.size) {
```

Comment:
is this really true?

Reply:
There's  a filter above, so it changes if and only if length doesn't match.